akka-stream/src/main/scala/akka/stream/scaladsl/FlattenStrategy.scala

```diff
@@ -15,7 +20,10 @@ object FlattenStrategy {
    * emitting its elements directly to the output until it completes and then taking the next stream. This has the
    * consequence that if one of the input stream is infinite, no other streams after that will be consumed from.
    */
-  def concat[T]: FlattenStrategy[scaladsl.Source[T, _], T] = Concat[T]()
+  def concat[T]: FlattenStrategy[Source[T, _], T] = Concat[T]()
 
-  private[akka] final case class Concat[T]() extends FlattenStrategy[scaladsl.Source[T, _], T]
+  private[akka] final case class Concat[T]() extends FlattenStrategy[Source[T, _], T] {
+    override def asJava[S]: javadsl.FlattenStrategy[S, T] =
```

Comment:
when trying to change corresponding thing in javadsl.FlattenStrategy to:

    def asScala: scaladsl.FlattenStrategy[scaladsl.Source[U, _], U]

compilation error in javadsl.Flow:

    def flatten[U](strategy: FlattenStrategy[Out, U]): javadsl.Flow[In, U, Mat] =
      new Flow(delegate.flatten(strategy.asScala))


    type mismatch; found : akka.stream.scaladsl.FlattenStrategy[akka.stream.scaladsl.Source[U, _],U] required: akka.stream.scaladsl.FlattenStrategy[Out,?]